spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts Controller', type: :request do
+  it 'GET users/id/posts should succeed' do
+    get '/users/id/posts'
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "Redirects to the user posts' page" do
+    get '/users/id/posts'
+    expect(response).to render_template(:index)
+    expect(response.body).to include('Comments: x, Likes: x')
+    expect(response.body).to include('Pagination')
+  end
+
+  it 'does not render a different template' do
+    get '/users/id/posts'
+    expect(response).to_not render_template(:show)
+  end
```

Comment:
- [ ] I would recommend wrapping the while index method test under one `describe` block and another `describe` block for show.

```ruby
RSpec.describe 'Posts Controller', type: :request do
  describe 'GET /index' do
    ...
  end
 
  describe 'GET /show' do
    ...
  end
end

```

Reply:
Hey @

Thank You for the feedback :pray: 
I really appreciate :+1: :+1: